### PR TITLE
<refactor> GEN3 directory tree to separate state

### PR DESCRIPTION
**Pull request by ml019**
This commit is the automation companion of the equivalent gen3 change.

The only specific change required for automation was to change build
promotion to use the new PRODUCT_BUILDS_DIR variable to locate where the
builds to be promoted should come from.

utility.sh and contextTree.sh have also been updated to pick up the
refactored directory location logic.

### CODE REVIEW

**Comment by ml019** [comment]
@roleyfoley This is ready for review.

File: contextTree.sh [hunk 1]
```diff
@@ -1187,21 +1751,24 @@ function process_cmdb() {
 
 function upgrade_cmdb() {
   local root_dir="$1";shift
+  local gen3_version="$1";shift
```
Comment (roleyfoley) [inline contextTree.sh]: Where do we get this from?

**Comment by roleyfoley** [comment]
Looks good, might need to be rebased
I think some of the changes you added for semver haven’t been picked up

File: contextTree.sh (hunk 1 above)
Comment (ml019) [reply contextTree.sh]: TBD - so far its not provided. It is only really needed if we need to constrain the update.

**Pull request merged**